src/Identity/v3/Models/Token.php

```diff
@@ -116,6 +119,22 @@ public function create(array $data): Creatable
         }
 
         $response = $this->execute($this->api->postTokens(), $data);
-        return $this->populateFromResponse($response);
+        $token = $this->populateFromResponse($response);
+
+        $this->cacheCredential = Utils::flattenJson(Utils::jsonDecode($response), $this->resourceKey);
+        $this->cacheCredential['id'] = $token->id;
+
+        return $token;
+    }
+
+    /**
+     * Retrieves an array serializable representation of authentication token.
+     * Can be use to initialise OpenStack object using $params['cachedCredential']
+     *
+     * @return array
+     */
+    public function exportCredential(): array
```

Comment:
this method seems pretty superfluous. how about we make `cachedToken` public and change this method to something like `exportToFile(string $path)` that persists to disk?

I think it's safe to make some assumption in a helper method. If people want to control their own caching implementation they can use `$token->cachedCredential` directly.

Reply:
I think `exportToFile` would be to specific. Another use case of this is caching the tokens in redis, memcache etc.

Reply:
Yes, but persistence to file is a common denominator scenario we can predict and offer a helper for.

Reply:
> persistence to file is a common denominator scenario

Hmm hard to make this assumption here since we don't have data to support. 

I tend to agree with @josiasmontag that SDK should not handle persistent data. `12-factors` methodology encourages apps not to store data locally but in a stateful back-end service.

Applications employ this SDK should decide where to store the token; it only takes a one-liner to write to file anyway.

Reply:
> how about we make `cachedToken` public 

Not a good idea.